src/main/java/com/gooddata/GoodDataEndpoint.java

```diff
@@ -62,7 +62,12 @@ public GoodDataEndpoint() {
      * @return the host URI, as a string.
      */
     public String toUri() {
-        return new HttpHost(hostname, port, protocol).toURI();
+        return UriComponentsBuilder
```

Comment:
This code is originally from `UriPrefixingClientHttpRequestFactory`. I used it because it looked better and also this way factory'll use same code for Uri creation.
But I can revert it back...